src/ui/public/visualize/visualization.js

```diff
@@ -118,10 +117,7 @@ uiModules
         visualization.render($scope.visData, status)
           .then(() => {
             // renderComplete
-            $scope.$emit('renderComplete');
-            if (!$scope.vis.visualizeScope) {
-              $el.trigger('renderComplete');
-            }
+            $el.trigger('renderComplete');
```

Comment:
we no longer trigger renderComplete on element ? 
using `<visualization>` we won't know when its done rendering ?

Reply:
Apologies, I thought the `$emit('renderComplete')` was only being utilized by the `render-counter` attribute, but it looks like it's being used a few other places, I'll put this logic back.

Reply:
Did the change expose a bug in the UI at all?  Wondering if we could add a test to catch the UI issues that it would have caused (assuming they passed...)

Reply:
there is no use of this as of now, this is there for 3rd party app developers